src/__tests__/scheduling/appointments/view/ViewAppointment.test.tsx

```diff
@@ -22,227 +25,195 @@ import { RootState } from '../../../../shared/store'
-    setButtonToolBarSpy = jest.fn()
-    jest.spyOn(ButtonBarProvider, 'useButtonToolbarSetter').mockReturnValue(setButtonToolBarSpy)
   })
 
   it('should have called the useUpdateTitle hook', async () => {
-    await setup()
+    setup()
 
-    expect(titleUtil.useUpdateTitle).toHaveBeenCalled()
+    await waitFor(() => {
+      expect(titleUtil.useUpdateTitle).toHaveBeenCalled()
+    })
   })
 
   it('should add a "Edit Appointment" button to the button tool bar if has WriteAppointment permissions', async () => {
-    await setup('loading', [Permissions.WriteAppointments, Permissions.ReadAppointments])
+    setup([Permissions.WriteAppointments])
 
-    const actualButtons: React.ReactNode[] = setButtonToolBarSpy.mock.calls[0][0]
-    expect((actualButtons[0] as any).props.children).toEqual('actions.edit')
+    await waitFor(() => {
+      expect(screen.getByRole('button', { name: /actions\.edit/i })).toBeInTheDocument()
+    })
   })
 
   it('should add a "Delete Appointment" button to the button tool bar if has DeleteAppointment permissions', async () => {
-    await setup('loading', [Permissions.DeleteAppointment, Permissions.ReadAppointments])
+    setup([Permissions.DeleteAppointment])
 
-    const actualButtons: React.ReactNode[] = setButtonToolBarSpy.mock.calls[0][0]
-    expect((actualButtons[0] as any).props.children).toEqual(
-      'scheduling.appointments.deleteAppointment',
-    )
+    await waitFor(() => {
+      expect(
+        screen.getByRole('button', { name: /scheduling\.appointments\.deleteAppointment/i }),
+      ).toBeInTheDocument()
+    })
   })
 
   it('button toolbar empty if has only ReadAppointments permission', async () => {
-    await setup('loading')
+    const { container } = setup()
 
-    const actualButtons: React.ReactNode[] = setButtonToolBarSpy.mock.calls[0][0]
-    expect(actualButtons.length).toEqual(0)
+    await waitFor(() => {
+      expect(container.querySelector(`[class^='css-']`)).not.toBeInTheDocument()
+    })
+
+    expect(screen.queryAllByRole('button')).toHaveLength(0)
   })
 
   it('should call getAppointment by id if id is present', async () => {
-    await setup()
-    expect(findAppointmentSpy).toHaveBeenCalledWith(appointment.id)
-  })
+    const { expectedAppointment } = setup()
 
-  it('should render a loading spinner', async () => {
-    const { wrapper } = await setup('loading')
-    expect(wrapper.find(components.Spinner)).toHaveLength(1)
+    expect(AppointmentRepository.find).toHaveBeenCalledWith(expectedAppointment.id)
   })
 
-  it('should render an AppointmentDetailForm with the correct data', async () => {
-    const { wrapper } = await setup()
-    wrapper.update()
+  // This relies on an implementation detial... Dunno how else to make it work
+  it('should render a loading spinner', () => {
+    // Force null as patient response so we get the "loading" condition
+    jest.spyOn(PatientRepository, 'find').mockResolvedValueOnce((null as unknown) as Patient)
```

Comment:
@JacobMGEvans I'm not sure how I feel about this. I couldn't find any other way around it, however...

Reply:
Well, unless we want to go the MSW route, then this is a good working solution.